libutils/json.h

```diff
@@ -354,13 +354,19 @@ JsonElement *JsonArrayGet(const JsonElement *array, size_t index);
   */
 bool JsonArrayContainsOnlyPrimitives(JsonElement *array);
 
+typedef JsonElement *JsonLookup(void *ctx, const char **data);
+
 /**
   @brief Parse a string to create a JsonElement
+  @param ctx [in] Evaluation context for variable lookups
   @param data [in] Pointer to the string to parse
   @param json_out Resulting JSON object
   @returns See JsonParseError and JsonParseErrorToString
+
+  The ctx type is void so we don't have to include eval_context.h from
+  libpromises into libutil
   */
-JsonParseError JsonParse(const char **data, JsonElement **json_out);
+JsonParseError JsonParse(void *ctx, JsonLookup *lookup, const char **data, JsonElement **json_out);
```

Comment:
OK; fixed. I renamed `ctx` to `lookup_context` because I think it's the best name for it, and `lookup` is now `lookup_function`. I hope that's acceptable, and you're right that the change is much smaller and cleaner that way.